packages/big-design/src/components/TableNext/HeaderCell/HeaderCell.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  ArrowDownwardIcon,
+  ArrowUpwardIcon,
+  BaselineHelpIcon,
+} from '@bigcommerce/big-design-icons';
+import React, { memo, RefObject, TableHTMLAttributes } from 'react';
+
+import { useComponentSize, useUniqueId } from '../../../hooks';
+import { typedMemo } from '../../../utils';
+import { Box } from '../../Box';
+import { Tooltip } from '../../Tooltip';
+import { TableColumnDisplayProps } from '../mixins';
+import { TableColumn, TableItem } from '../types';
+
+import { StyledFlex, StyledTableHeaderCell, StyledTableHeaderIcon } from './styled';
+
+export interface HeaderCellProps<T>
+  extends TableHTMLAttributes<HTMLTableCellElement>,
+    TableColumnDisplayProps {
+  actionsRef: RefObject<HTMLDivElement>;
+  children?: React.ReactNode;
+  column: TableColumn<T>;
+  id: string;
+  hide?: boolean;
+  isSorted?: boolean;
+  sortDirection?: 'ASC' | 'DESC';
+  stickyHeader?: boolean;
+  onSortClick?(column: TableColumn<T>): void;
+}
+
+export interface HeaderCheckboxCellProps {
+  actionsRef: RefObject<HTMLDivElement>;
+  stickyHeader?: boolean;
+}
+
+export interface DragIconCellProps {
+  actionsRef: RefObject<HTMLDivElement>;
+  headerCellIconRef: RefObject<HTMLTableCellElement>;
+}
+
+const InternalHeaderCell = <T extends TableItem>({
+  actionsRef,
+  children,
+  column,
+  display,
+  hide = false,
+  id,
+  isSorted,
+  onSortClick,
+  sortDirection,
+  stickyHeader,
+}: HeaderCellProps<T>) => {
+  const { align = 'left', isSortable, width, tooltip } = column;
+  const actionsSize = useComponentSize(actionsRef);
+  const tooltipId = useUniqueId('table-header-tooltip');
+
+  const renderSortIcon = () => {
+    if (!isSorted) {
+      return null;
+    }
+
+    return sortDirection === 'ASC' ? (
+      <ArrowUpwardIcon data-testid="asc-icon" size="medium" title="Ascending order" />
+    ) : (
+      <ArrowDownwardIcon data-testid="desc-icon" size="medium" title="Descending order" />
+    );
+  };
+
+  const renderTooltip = () => {
+    if (typeof tooltip === 'string' && tooltip.length > 0) {
+      return (
+        <Tooltip
+          id={tooltipId}
+          placement="right"
+          trigger={
+            <Box as="span" marginLeft="xxSmall">
+              <BaselineHelpIcon
+                aria-describedby={tooltipId}
+                size="medium"
+                title="Hover or focus for additional context."
```

Comment:
`title` need `getText`?

Reply:
Good call out! I think we don't translate texts in BD for the moment, however, this is one of our future plans. If I recall correctly the Design team created a doc talking about this. So might be good to wait until we decided what's the next step?

Reply:
Good callout, but yeah lets not worry too much about gettext for now